# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: 4e0758ff6af7
# Full Hash: 4e0758ff6af7d0cdd1d8a2c6c8dec21ba1bd4ab9
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-12-05 04:11:08
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Make use of FlippedOnce in DatabaseInfo. r=dom-workers-and-storage-reviewers,ytausky
#   
#   Depends on D55679
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D55680
# ==============================================================================

diff -r dc96d3709a42 -r 4e0758ff6af7 dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Wed Dec 04 14:22:33 2019 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Wed Dec 04 14:21:51 2019 +0000
@@ -5134,7 +5134,7 @@
   uint32_t mWriteTransactionCount;
   bool mNeedsCheckpoint;
   bool mIdle;
-  bool mCloseOnIdle;
+  FlippedOnce<false> mCloseOnIdle;
   bool mClosing;
 
 #ifdef DEBUG
@@ -12010,7 +12010,7 @@
       CloseDatabase(dbInfo);
       AdjustIdleTimer();
     } else {
-      dbInfo->mCloseOnIdle = true;
+      dbInfo->mCloseOnIdle.EnsureFlipped();
     }
 
     return true;
@@ -12120,7 +12120,6 @@
       mWriteTransactionCount(0),
       mNeedsCheckpoint(false),
       mIdle(false),
-      mCloseOnIdle(false),
       mClosing(false)
 #ifdef DEBUG
       ,
